### PR TITLE
New build process and folder renaming among other changes

### PR DESCRIPTION
**Pull request by Ben-m-s**
This PR includes the following major changes:
  - New PowerShell module to manage the build process for docker images
  - Update build.json files to support the new build process and new assets management process.
  - Folder renaming based on Sitecore and Docker naming conventions
  - Optimized layers in images
  - New XP Single image with Sitecore 9.0.1 and 9.0.2

### CODE REVIEW

**Comment by Ben-m-s** [comment]
Hi all,
I have added a file with CREDITS consisting of rough summaries of the contributions made by the people I could think of. I may have missed people or relevant aspects to be highlighted. I apologise in advance. Additions and corrections are welcome.
The README.md file has been updated with some guidelines about the new changes. Feedback is welcome.

File: CREDITS.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+**Per Bering**: One of the first promoters of Docker for Sitecore. He is a major contributor to this repository, and actively helps peers with Docker related matters on different channels.
+https://github.com/sitecoreops/sitecore-images
+https://github.com/Sitecore/docker-demo
+https://github.com/pbering/SitecoreContainers
+https://github.com/sitecoreops/sitecore-images
+https://github.com/pbering/sitecore-nine-docker
+https://github.com/pbering/SockerVS2017
+https://github.com/pbering/solrwin
+
+**Joost Meijles (Aviva Solutions)**: His contributions in GitHub have been a source of inspiration. He is also active in community discussions about using Docker with Sitecore.
```
Comment (joostmeijles) [inline CREDITS.md]: Please:
- add https://github.com/avivasolutionsnl/sitecore-commerce-docker iso https://github.com/joostmeijles/sitecore-commerce-docker
- remove my personal fork (and not maintained): https://github.com/joostmeijles/sitecore-docker

Comment (Ben-m-s) [reply CREDITS.md]: Done

File: CREDITS.md [hunk 2]
```diff
@@ -12,10 +12,11 @@ https://github.com/pbering/SockerVS2017
 https://github.com/pbering/solrwin
 
 **Joost Meijles (Aviva Solutions)**: His contributions in GitHub have been a source of inspiration. He is also active in community discussions about using Docker with Sitecore.
-https://github.com/sitecoreops/sitecore-images
+https://github.com/avivasolutionsnl/sitecore-commerce-docker
 https://github.com/joostmeijles/sitecore-commerce-docker
```
Comment (joostmeijles) [inline CREDITS.md]: Please remove this https://github.com/joostmeijles/sitecore-commerce-docker link (its only a not-well maintained fork of https://github.com/avivasolutionsnl/sitecore-commerce-docker)

File: CREDITS.md [hunk 3]
```diff
@@ -12,10 +12,11 @@ https://github.com/pbering/SockerVS2017
 https://github.com/pbering/solrwin
 
 **Joost Meijles (Aviva Solutions)**: His contributions in GitHub have been a source of inspiration. He is also active in community discussions about using Docker with Sitecore.
```
Comment (joostmeijles) [inline CREDITS.md]: Please add Jonne Kats (he is the Jonne below) here as well. His efforts are a big as mine for the avivasolutionsnl docker repos. Thanks.

Comment (Ben-m-s) [reply CREDITS.md]: Done

**Pull request merged**